src/preview.ts

```diff
@@ -0,0 +1,178 @@
+                localResourceRoots: [vscode.Uri.joinPath(extensionUri, 'media')]
+            }
+        );
+
+        PreviewPanel.currentPanel = new PreviewPanel(panel, extensionUri);
+    }
+
+    public static revive(panel: vscode.WebviewPanel, extensionUri: vscode.Uri) {
+        PreviewPanel.currentPanel = new PreviewPanel(panel, extensionUri);
+    }
+
+    private constructor(panel: vscode.WebviewPanel, extensionUri: vscode.Uri) {
+        this._panel = panel;
+        this._extensionUri = extensionUri;
+
+        // Set the webview's initial html content
+        this._document = vscode.window.activeTextEditor.document;
+        this._update();
+
+        // Update the webview based on text changes
+        vscode.workspace.onDidChangeTextDocument(
+            e => {
+                if (e.document.uri.toString() === this._document.uri.toString()) {
+                    this._update();
+                }
+            },
+            null,
+            this._disposables);
+
+
+        // Update when the user switches to a new editor
+        vscode.window.onDidChangeActiveTextEditor(
+            e => { 
+                this._document = e.document;
+                this._update();
+            },
+            null,
+            this._disposables);
+
+        // Listen for when the panel is disposed
+        // This happens when the user closes the panel or when the panel is closed programatically
+        this._panel.onDidDispose(() => this.dispose(), null, this._disposables);
+
+        // Update the content based on view changes
+        this._panel.onDidChangeViewState(
+            () => {
+                if (this._panel.visible) {
+                    this._update();
+                }
+            },
+            null,
+            this._disposables
+        );
+
+        // Handle messages from the webview
+        this._panel.webview.onDidReceiveMessage(
+            message => {
+                switch (message.command) {
+                    case 'alert':
+                        vscode.window.showErrorMessage(message.text);
+                        return;
+                }
+            },
+            null,
+            this._disposables
+        );
+    }
+
+    public doRefactor() {
+        // Send a message to the webview webview.
+        // You can send any JSON serializable data.
+        this._panel.webview.postMessage({ command: 'refactor' });
+    }
+
+    public dispose() {
+        PreviewPanel.currentPanel = undefined;
+
+        // Clean up our resources
+        this._panel.dispose();
+
+        while (this._disposables.length) {
+            const x = this._disposables.pop();
+            if (x) {
+                x.dispose();
+            }
+        }
+    }
+
+    private _update() {
+        const webview = this._panel.webview;
+        this._updateForPreview(webview);
+    }
+
+    private _updateForPreview(webview: vscode.Webview) {
+        this._panel.title = "Oracc Preview";
```

Comment:
When I renamed things I missed this line: a few lines below, in `_getHtmlForWebview`, I titled the webpage "Nisaba Preview" which is maybe a bit more accurate?  I don't care too much which one we pick, but I guess we should be consistent